### PR TITLE
explicitly mark nullable parameters as nullable

### PR DESCRIPTION
**Pull request by xabbuh**
not doing this will be deprecated in PHP 8.4 (see e.g. https://github.com/symfony/symfony/actions/runs/8459191951/job/23175036483?pr=54426#step:8:2335)

### CODE REVIEW

**Comment by xabbuh** [comment]
@tillkruss done, should I also fix the reported code style issues?

**Comment by tillkruss** [comment]
In a separate PR, sure! 👍

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -69,7 +70,7 @@ jobs:
         run: vendor/bin/phpunit --coverage-clover build/logs/clover-default.xml --coverage-filter ./src
 
       - name: Run tests using Relay
-        if: ${{ matrix.redis >= '6' }}
+        if: ${{ matrix.php != '8.4' && matrix.redis >= '6' }}
```
Comment (tillkruss) [inline .github/workflows/tests.yml:73]: Is this because there are no 8.4 builds for Relay?

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (xabbuh) [reply .github/workflows/tests.yml:73]: Yes, the tests were failing because the extension couldn’t be installed.

**Comment by nicolas-grekas** [comment]
Any blocker to merge this? That's remove some failures in the CI of Symfony. :pray:

**Comment by nicolas-grekas** [comment]
@tillkruss sorry for the noise. Predis is the last dep of Symfony that didn't fix this issue. I'd love being able to close this topic :pray:

**Comment by nicolas-grekas** [comment]
@tillkruss is there anything we could do to help get this PR merged? :pray:

**Comment by tillkruss** [comment]
@nicolas-grekas Yes, sorry for the delay.

**Comment by xabbuh** [comment]
Thank you for merging! 😊

**Comment by kenjis** [comment]
When will this release?

**Comment by crynobone** [comment]
@tillkruss Any chance we can have `2.2.3` release this week as PHP 8.4 GA is planned for November 21st, 2024 (this thursday).

**Comment by tillkruss** [comment]
@crynobone Yeah good idea, heaps of changes queued up: https://github.com/predis/predis/compare/v2.2.2...v2.x

**Comment by tillkruss** [comment]
@crynobone v2.3.0 was tagged https://github.com/predis/predis/releases/tag/v2.3.0

**Pull request merged**